src/Workspaces/Core/Portable/Classification/ClassifierHelper.cs

```diff
@@ -45,12 +45,17 @@ internal static partial class ClassifierHelper
             using var _1 = Classifier.GetPooledList(out var syntaxSpans);
             using var _2 = Classifier.GetPooledList(out var semanticSpans);
 
-            await classificationService.AddSyntacticClassificationsAsync(document, span, syntaxSpans, cancellationToken).ConfigureAwait(false);
+            for (var i = 0; i < spans.Length; i++)
+            {
+                var span = spans[i];
+                await classificationService.AddSyntacticClassificationsAsync(document, span, syntaxSpans, cancellationToken).ConfigureAwait(false);
 
-            // Intentional that we're adding both semantic and embedded lang classifications to the same array.  Both
-            // are 'semantic' from the perspective of this helper method.
-            await classificationService.AddSemanticClassificationsAsync(document, span, options, semanticSpans, cancellationToken).ConfigureAwait(false);
-            await classificationService.AddEmbeddedLanguageClassificationsAsync(document, span, options, semanticSpans, cancellationToken).ConfigureAwait(false);
+                // Intentional that we're adding both semantic and embedded lang classifications to the same array.  Both
+                // are 'semantic' from the perspective of this helper method.
+                // AddSemanticClassificationsAsync is part of a hotpath, could we send down a text span of the full document instead?
+                await classificationService.AddSemanticClassificationsAsync(document, span, options, semanticSpans, cancellationToken).ConfigureAwait(false);
```

Comment:
so this will continue to be a problem.  because we'll make a call to oop for each span, and each call will create a semantic model that will perform all the computation and then throw it away.  all teh spans needs to be pushed through all the way through here.